server/methods/deleteMessage.coffee

```diff
@@ -18,7 +17,7 @@ Meteor.methods
 
 		deleteQuery = 
 			_id: message._id
-		deleteQuery['u._id'] = Meteor.userId() if user?.admin isnt true 
+		#deleteQuery['u._id'] = Meteor.userId() if user?.admin isnt true 
```

Comment:
ok for `unless hasPermission or (deleteAllowed and deleteOwn)`, the problem is `deleteOwn = message?.u?._id is Meteor.userId()` that is believing in the user passed to the method.